src/objective-c/RxLibrary/GRXForwardingWriter.m

```diff
@@ -48,7 +48,10 @@ - (instancetype)init {
 // Designated initializer
 - (instancetype)initWithWriter:(GRXWriter *)writer {
   if (!writer) {
-    [NSException raise:NSInvalidArgumentException format:@"writer can't be nil."];
+    return nil;
+  }
+  if (writer.state != GRXWriterStateNotStarted) {
+    [NSException raise:NSInvalidArgumentException format:@"writer can't be started."];
```

Comment:
The error text is hard to understand. better:
"writer argument must not have already started."